.github/workflows/_test_coverage.yml

```
@@ -0,0 +1,131 @@
+          go tool cover -html=coverage/${{ inputs.output_name }}.coverage -o coverage/${{ inputs.output_name }}.html
+          
+          # Generate functional coverage summary
+          go tool cover -func=coverage/${{ inputs.output_name }}.coverage | tee coverage/${{ inputs.output_name }}-summary.txt
+          
+          # Extract total coverage percentage for the job summary
+          TOTAL_COV=$(go tool cover -func=coverage/${{ inputs.output_name }}.coverage | grep total | awk '{print $3}')
+          echo "total_coverage=$TOTAL_COV" >> $GITHUB_OUTPUT
+
+      - name: Upload to Codecov
+        if: inputs.codecov
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.4)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)